server/index.js

```diff
@@ -331,9 +331,16 @@ app.get("/_flaws", (req, res) => {
           ((b.popularity.value || 0) - (a.popularity.value || 0))
         );
       case "flaws":
-        // This is kinda bogus and slow.
-        const vA = a.flaws.reduce((x, y) => x + y.value, 0);
-        const vB = b.flaws.reduce((x, y) => x + y.value, 0);
+        const vA = a.flaws
+          .filter(({ name }) => {
+            return !filteredFlaws.size || filteredFlaws.has(name);
+          })
+          .reduce((x, y) => x + y.value, 0);
+        const vB = b.flaws
+          .filter(({ name }) => {
+            return !filteredFlaws.size || filteredFlaws.has(name);
+          })
+          .reduce((x, y) => x + y.value, 0);
```

Comment:
I'd do some extracting here, mostly for clarity but dedup is a nice extra. E.g.:

```javascript
// somewhere before .sort()
function countFilteredFlaws(doc) {
  return doc.flaws
    .filter(({ name }) => filteredFlaws.size || filteredFlaws.has(name))
    .reduce((x, y) => x + y.value, 0);
}

// inside case "flaws":
return sortMultiplier * (countFilteredFlaws(b) - countFilteredFlaws(a));
```

Reply:
Got it.